File: src/JsonServices.Core/Auth/AuthRequest.cs
using System.Collections.Generic;
using JsonServices.Messages;

namespace JsonServices.Auth
{
	public class AuthRequest : IReturn<AuthResponse>, ICustomName
	{
		public const string MessageName = "rpc.authenticate";

		string ICustomName.MessageName => MessageName;

		public const string UserNameKey = "UserName";

		public const string PasswordKey = "Password";

		public Dictionary<string, string> Parameters { get; set; } =
			new Dictionary<string, string>();
	}
}

File: sample/sample-client/src/JsonServices/AuthRequest.ts
import { AuthResponse } from './AuthResponse';
import { IReturn } from './IReturn';

export class AuthRequest implements IReturn<AuthResponse> {
    public static userNameKey = "UserName";
    public static passwordKey = "Password";
    public getTypeName = () => "rpc.authenticate";
    public createResponse = () => new AuthResponse();
    public Parameters: {
        [key: string]: string;
    } = {};
}